### PR TITLE
implement memory backend backup

### PR DESCRIPTION
**Pull request by zhoney**
Change-Id: Ie5d6e9306df9d821c2f7bdae0cd8face6e984a98

### CODE REVIEW

File: src/main/java/com/baidu/hugegraph/manager/BackupManager.java [hunk 1]
```diff
@@ -281,6 +294,10 @@ private void backup(HugeType type, int number, List<?> list) {
         }
     }
 
+    private String initPage() {
+        return this.backend.equals(MEMORY) ? null : "";
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/manager/BackupManager.java]: can keep it consistent?

**Review by javeme — CHANGES_REQUESTED** [review]

File: src/main/java/com/baidu/hugegraph/manager/BackupManager.java (hunk 1 above)
Comment (zhoney) [reply src/main/java/com/baidu/hugegraph/manager/BackupManager.java]: memory is special due to not support paging

File: src/main/java/com/baidu/hugegraph/manager/BackupManager.java [hunk 2]
```diff
@@ -67,14 +67,19 @@
 
     public static final int BACKUP_DEFAULT_TIMEOUT = 120;
 
+    private static final String BACKEND = "backend";
+    private static final String MEMORY = "memory";
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/manager/BackupManager.java]: rename to BACKENDS_NO_PAGING

**Review by javeme — DISMISSED** [review]

**Review by houzhizhen — DISMISSED** [review]

File: src/main/java/com/baidu/hugegraph/manager/BackupManager.java [hunk 3]
```diff
@@ -56,6 +57,7 @@
 import com.baidu.hugegraph.util.E;
 
 import jersey.repackaged.com.google.common.collect.ImmutableList;
+import jersey.repackaged.com.google.common.collect.ImmutableSet;
```
Comment (Linary) [inline src/main/java/com/baidu/hugegraph/manager/BackupManager.java]: can use `com.google.common.collect.ImmutableSet`?

**Review by Linary — CHANGES_REQUESTED** [review]

**Review by Linary — APPROVED** [review]

**Review by houzhizhen — APPROVED** [review]

**Review by houzhizhen — APPROVED** [review]

**Review by houzhizhen — APPROVED** [review]

**Review by javeme — APPROVED** [review]

**Pull request merged**